### ISSUE TITLE
after pull rand() script not work,

### ISSUE THREAD

**Issue by dzurrahman** [rathena/rathena#8041]
### rAthena Hash

e64b1cc23

### Client Date

2022-03-30

### Server Mode

Renewal

### Result

set .@indx,rand(.MaxIndex);

not work ..

npc script

rand result awalys 0

### Relevant Log Output

```Shell
-
```

### Expected Result

-

### How to Reproduce

-

### Official Information

-

### Modifications that may affect results

-

**Comment by Lemongrass3110** [comment]
Can you share your script?
What is the value of `.MaxIndex`?

I am unable to reproduce this.

```
prontera,155,177,4	script	RandTest	100,{
	for( .@i = 20; .@i <= 40; .@i++ ){
		.@tmp$ = "";

		for( .@j = 0; .@j < 5; .@j++ ){
			if( .@j > 0 ){
				.@tmp$ += ",";
			}

			.@tmp$ += rand ( .@i );
		}

		mes "rand(" + .@i + "): " + .@tmp$;
	}

	close;
}
```

![grafik](https://github.com/rathena/rathena/assets/3517879/4237aa62-4f96-4aa7-a136-c4c7a4b80c0f)
![grafik](https://github.com/rathena/rathena/assets/3517879/e5060600-c650-438e-b1d3-79c220a3d4b7)
![grafik](https://github.com/rathena/rathena/assets/3517879/09bf58cc-d03b-4db3-982e-71eabfd4c87f)

**Comment by dzurrahman** [comment]
![image](https://github.com/rathena/rathena/assets/56845420/7d995e98-68e2-4c8d-bb43-b6ce76f422d6)
![image](https://github.com/rathena/rathena/assets/56845420/9c162c29-554f-483c-9bb6-4d72e0598f21)

```
prontera,155,177,4	script	RandTest	100,{
    set .MaxIndex,2;
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    mes rand(.MaxIndex);
    next;
    mes rand(2);
    mes rand(2);
    mes rand(2);
    mes rand(2);
    mes rand(2);
    mes rand(2);
    mes rand(2);
    mes rand(2);
    mes rand(2);
    mes rand(2);
    mes rand(2);
    mes rand(2);
    mes rand(2);
    mes rand(2);
    mes rand(2);
	close;
}
```

**Comment by Lemongrass3110** [comment]
What OS?
Which compiler?
Which compiler version?

**Comment by dzurrahman** [comment]
OS: Ubuntu 18.04.6 LTS x86_64
Kernel: 4.15.0-213-generic
CPU: Intel Xeon E5-2690 v4 (4) @ 2.599GHz

gcc --version
gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE

**Comment by Lemongrass3110** [comment]
This problem is fixed in #8040

**Comment by dzurrahman** [comment]
after pull this merger, rand(1) or rand(0) not work!!!
